### PR TITLE
Add the UMF_BUILD_BENCHMARKS option

### PR DESCRIPTION
**Pull request by ldorau**
Add the `UMF_BUILD_BENCHMARKS` option and a new workflow for running benchmarks.

### CODE REVIEW

File: CMakeLists.txt [hunk 1]
```diff
@@ -75,8 +76,12 @@ add_subdirectory(src)
 if(UMF_BUILD_TESTS)
     add_subdirectory(test)
 endif()
-if(LINUX)
-    add_subdirectory(benchmark)
+if(UMF_BUILD_BENCHMARKS)
+    if(LINUX)
+        add_subdirectory(benchmark)
+    else()
+        message(FATAL_ERROR "benchmarks are supported on Linux only")
```
Comment (igchor) [inline CMakeLists.txt:83]: What's the reason behind this? Is ubench only supported on linux?

**Review by igchor — APPROVED** [review]

File: CMakeLists.txt (hunk 1 above)
Comment (ldorau) [reply CMakeLists.txt:83]: OS memory provider is not implemented for Windows.

**Comment by ldorau** [comment]
Add the `UMF_BUILD_BENCHMARKS` option and a new workflow for running benchmarks.

File: .github/workflows/benchmarks.yml [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+          -B ${{github.workspace}}/build
+          -DCMAKE_BUILD_TYPE=${{matrix.build_type}}
+          -DCMAKE_C_COMPILER=${{matrix.compiler.c}}
+          -DCMAKE_CXX_COMPILER=${{matrix.compiler.cxx}}
+          -DUMF_BUILD_BENCHMARKS=ON
+          -DUMF_BUILD_TESTS=OFF
+          -DUMF_BUILD_LIBUMF_POOL_JEMALLOC=OFF
+          -DUMF_FORMAT_CODE_STYLE=OFF
+          -DUMF_DEVELOPER_MODE=OFF
+          -DUMF_BUILD_LIBUMF_POOL_JEMALLOC=OFF
```
Comment (bratpiorka) [inline .github/workflows/benchmarks.yml:37]: You set UMF_BUILD_LIBUMF_POOL_JEMALLOC twice. BTW - shouldn't we add jemalloc to benchmark also?

**Review by bratpiorka — CHANGES_REQUESTED** [review]

File: .github/workflows/benchmarks.yml (hunk 2 above)
Comment (ldorau) [reply .github/workflows/benchmarks.yml:36]: Yes, but in a separate PR.
Done.

**Review by bratpiorka — APPROVED** [review]

**Pull request merged**